server/sheet-data.js

```diff
@@ -5,96 +5,319 @@ const config = require('./configuration');
+    // Who Found This?
+    data.user,
+    // 1
+    annotation.indiv_1 ? 'y' : '',
+    // 2
+    annotation.indiv_2 ? 'y' : '',
+    // 3
+    annotation.indiv_3 ? 'y' : '',
+    // 4
+    annotation.indiv_4 ? 'y' : '',
+    // 5
+    annotation.indiv_5 ? 'y' : '',
+    // 6
+    annotation.indiv_6 ? 'y' : '',
+    // 7
+    annotation.repeat_7 ? 'y' : '',
+    // 8
+    annotation.repeat_8 ? 'y' : '',
+    // 9
+    annotation.repeat_9 ? 'y' : '',
+    // 10
+    annotation.repeat_10 ? 'y' : '',
+    // 11
+    annotation.repeat_11 ? 'y' : '',
+    // 12
+    annotation.repeat_12 ? 'y' : '',
+    // 1
+    annotation.sig_1 ? 'y' : '',
+    // 2
+    annotation.sig_2 ? 'y' : '',
+    // 3
+    annotation.sig_3 ? 'y' : '',
+    // 4
+    annotation.sig_4 ? 'y' : '',
+    // 5
+    annotation.sig_5 ? 'y' : '',
+    // 6
+    annotation.sig_6 ? 'y' : '',
+    // Choose from drop down menu (2 columns)
+    '', '',
+    // Leave blank (used on Patterns sheet)
+    '',
+    // Further Notes
+    annotation.notes || ''
+  ];
 
-    for (var i = 0; i < 5; i++) {
-        text += possible.charAt(Math.floor(Math.random() * possible.length));
+  return appendRowToSheet(
+    row,
+    config.baseConfiguration().GOOGLE_IMPORTANT_CHANGE_SHEET_ID,
+    'A6:AN6'
+  )
+    .then(() => 'Successfully appended');
+}
+
+/**
+ * Append a row to a google sheet
+ *
+ * @param {string[]} values Column values of row  to append
+ * @param {string}   spreadsheedId ID of Google sheet to append to
+ * @param {string}   [range] Range identifying the data to append to in sheet,
+ *   e.g. `A3:Z3`. This is NOT the range after which to append -- Google sheets
+ *   identifies a continuous series of rows that intersect with this range and
+ *   appends *after those rows.*
+ * @returns {Promise<Object>} Response data from Google Sheets
+ */
+function appendRowToSheet(values, spreadsheetId, range = 'A3:ZZZ') {
+  return addAuthentication({
+    spreadsheetId,
+    // supply a cell where data exists, Google decides for itself where the data table ends and appends, using extreme range again to grab everything
+    range,
+    resource: {
+      values: [values]
+    },
+    valueInputOption: 'RAW',
+    insertDataOption: 'INSERT_ROWS'
+  })
+    .then(promisable(sheets.spreadsheets.values.append));
+}
+
+
+// ------------------- PRIVATE UTILITIES -----------------------
+
+let authTokens;
+let authClient;
+
+/**
+ * Add authentication information to a request object for Google API calls.
+ * If there is an authenticated client on hand, it will be used. Otherwise,
+ * a new authenticated client will be created and logged in before returning
+ * the object with authentication information.
+ * @private
+ * @param {Object} requestData
+ * @returns {Promise<Object>} requestData modified with auth client
+ */
+function addAuthentication (requestData) {
+  return new Promise((resolve, reject) => {
+    if (!authTokens || Date.now() > authTokens.expiry_date - 30000) {
```

Comment:
We aren’t checking every 30 seconds—we’re checking whenever a request is made (which is the right thing to do with `expires`, unless we had a more complicated system that sets a timer to automatically throw out/renew tokens). The 30 seconds is a leeway—it’s saying not to use the token if it’s within 30 seconds of expiring (because a) our clock is not guaranteed to match Google’s and b) a request could potentially hang for a while).

Not sure I’m following with the `const` concern—which variable are you thinking should be a constant?